src/olympia/discovery/utils.py

```diff
@@ -0,0 +1,45 @@
+import json
+
+from django.conf import settings
+
+import requests
+from django_statsd.clients import statsd
+
+import olympia.core.logger
+from olympia import amo
+from olympia.addons.models import Addon
+
+from . import data
+
+
+log = olympia.core.logger.getLogger('z.amo')
+
+
+def call_recommendation_server(telemetry_id, endpoint):
+    log.debug(u'Calling recommendation server: {0}'.format(endpoint))
+    with statsd.timer('services.recommendations'):
+        response = requests.post(
+            endpoint,
+            timeout=settings.SIGNING_SERVER_TIMEOUT)

```

Comment:
Need a different setting.

In this case I think we want a very very short timeout in order to fall back on not doing replacements if the recommendation server has issues responding.

Reply:
Also, you need to catch `RequestException` to handle connection errors, timeouts etc

Reply:
how short btw?  5 seconds?  1 second?

Reply:
https://github.com/mozilla/addons-server/pull/6324/commits/5d695e8ac12962936d1e3bfedbbe321ad5d2709f adds the exception checking and sets the timeout to 5 seconds.

Reply:
I'd say even lower than that, like `0.1` (10 ms). But we'll have to test and tweak it.

Reply:
https://github.com/mozilla/addons-server/pull/6324/commits/acad75d56550ee9502081a263043c3176f1e340d changes it to 1 second* and makes the settings settable in env so we can tweak without pushing a commit each time.

* setting a timeout for an external service to less than a second seems very optimistic to me.  We only have a shorter timeout for localhost urls.

Reply:
I'm fine with 1 second to start testing, but IMHO we should be able to answer under a second for the whole endpoint.